components/auth.tsx

```diff
@@ -1,25 +1,28 @@
+import { UserContext } from "@/context/userContext"
 import { useSession } from "next-auth/react"
 import { useRouter } from "next/router"
-import { ReactNode, useEffect } from "react"
+import { ReactNode, useContext, useEffect } from "react"
 
 export const Auth: React.FC<{
   children: ReactNode
   role: number
   redirect: string
 }> = ({ children, role, redirect }) => {
   // if `{ required: true }` is supplied, `status` can only be "loading" or "authenticated"
-  const { status, data } = useSession({ required: true })
+  const { status } = useSession({ required: true })
+  const { user } = useContext(UserContext)
```

Comment:
edited the auth component to rely on the user context as its source of truth for the user's role instead of next-auth session. this is more manageable, as we can update the user context without logging out